Adds external package "oracle"-support for Elm 0.19

r1sc opened the pull request:
This PR aims to "fix" issue #260. A new "oracle" (this time from Delphi!) spits out elm-oracle-compatible documentation-info from the package cache for Elm version 0.19.

Requires the packages to be in the cache. Assumed to be located under ~/.elm/0.19.0/package or %appdata%/elm/0.19.0/package

r1sc commented:
Obligatory screenshot of a 0.19 project loaded.

![image](https://user-images.githubusercontent.com/629669/44949917-94598b80-ae3b-11e8-9511-8294db839624.png)

hakonrossebo commented:
Great work! I verified that this works fine on Windows 10. On my Ubuntu 18.04, it fails on this line
```
  let packageFolder = path.join(process.env.ELM_HOME || path.join(process.env.appdata, "elm"), "0.19.0/package");
```
Both ELM_HOME and process.env.appdata are undefined. Not sure if I have time to investigate alternatives right now. 
One other thing. There are linting errors from TSLint - please check with your file open and press F6/Ctrl-Shift-b to build.

hakonrossebo commented:
I guess you could use utils.isWindows and use ~/.elm/0.19 for Linux/

r1sc commented:
Great! I’ll fix the paths and the linter errors tonight. 👍

r1sc commented:
I used the OS-detection from elmUtils as you mentioned and changed the path detection to
```
const elmRoot = windowsOS ? path.join(process.env.appdata, 'elm') : '~/elm';
const packageFolderPath = path.join(elmRoot, '0.19.0/package');
```
This should work, but I have no way of testing it since I'm on Windows myself.

Also took care of the linter errors and squashed the commits into one nice package.

hakonrossebo commented:
Got it to work, but I had to use another approach to get the correct home directory
```
let os = require('os'); // at the top
//and this
  const elmRoot = windowsOS ? path.join(process.env.appdata, 'elm') : path.join(os.homedir(), '.elm'); //remember the . before elm here
```
You can probably clean up the last line, as this should work similar on all systems.
https://stackoverflow.com/questions/44116983/how-can-i-create-a-directory-in-user-home-using-nodejs

r1sc commented:
Ok the changes are in, I was able to clean it up a bit:

`const elmRoot = path.join(windowsOS ? process.env.appdata : os.homedir(), 'elm');`

hakonrossebo commented:
Ok, I have not verified, but on linux, the elm folder is in a dotfile folder - you need to prefix with a . - like '.elm'

r1sc commented:
Fixed

hakonrossebo commented:
Thanks. I can now confirm that this works fine on Windows 7/10 and Ubuntu and ok to merge PR.

Krzysztof-Cieslak commented:
Amazing, thanks a lot!
